tests/plugins/test_expert_hybrid.py

```
@@ -47,6 +117,89 @@ async def query(self, collection, query_texts, n_results=10, where=None):
+    import contextvars
+    from core.adapters.chromadb import ChromaDBAdapter
+
+    adapter = ChromaDBAdapter.__new__(ChromaDBAdapter)
+    adapter._query_embedding_cache = contextvars.ContextVar("traced-expert-scope", default=None)
+    outer_calls = 0
+    provider_attempts = 0
+    first_attempt = __import__("asyncio").Event()
+    release_retry = __import__("asyncio").Event()
+    class Embeddings:
+        async def embed_query(self, texts):
+            nonlocal outer_calls, provider_attempts
```

Flake8 (7.3.0)

[error] `nonlocal provider_attempts` is unused: name is never assigned in scope

(F824)